ddtrace/llmobs/_llmobs.py

```diff
@@ -230,16 +230,23 @@ def disable(cls) -> None:
         log.debug("%s disabled", cls.__name__)
 
     @classmethod
-    def annotation_context(cls, tags: Optional[Dict[str, Any]] = None) -> AnnotationContext:
+    def annotation_context(
+        cls, tags: Optional[Dict[str, Any]] = None, prompt: Optional[dict] = None, name: Optional[str] = None
+    ) -> AnnotationContext:
         """
         Sets specified attributes on all LLMObs spans created while the returned AnnotationContext is active.
-        Do not use nested annotation contexts to override the same tags since the order in which annotations
+        Do not use nested annotation contexts to override the same attributes since the order in which annotations
         are applied is non-deterministic.
```

Comment:
Why is it non-deterministic?

Reply:
We register the annotation function as a hook on span start. Because these [hooks are stored as a set](https://github.com/DataDog/dd-trace-py/blob/5546c2391097103c74688c66b0dd57f985194d8f/ddtrace/_hooks.py#L79C1-L80C1), the order in which these hooks are called is non-deterministic